results/cmd/api/main_test.go

```diff
@@ -54,12 +57,42 @@ func setupTestDB(dbName string, t *testing.T) (*sql.DB, error) {
 	})
 
 	// Create taskrun table
-	statement, err := db.Prepare("CREATE TABLE IF NOT EXISTS taskrun (logid binary(16) PRIMARY KEY, taskrunlog BLOB, uid INTEGER, name TEXT, namespace TEXT)")
+	statement, err := db.Prepare("CREATE TABLE IF NOT EXISTS taskrun (logid binary(16) PRIMARY KEY, taskrunlog BLOB, uid TEXT, name TEXT, namespace TEXT)")
 	if err != nil {
 		t.Fatalf("failed to create taskrun table: %v", err)
 	}
 	if _, err := statement.Exec(); err != nil {
 		t.Fatalf("failed to execute the taskrun table creation statement statement: %v", err)
 	}
-	return db, nil
+	return srv, nil
+}
+
+func TestGetTaskRun(t *testing.T) {
+	// Create a temporary database
+	srv, err := setupTestDB("testdb", t)
+	if err != nil {
+		t.Fatalf("failed to setup db: %v", err)
+	}
+
+	// Connect to fake server and create a new taskrun
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
+	r, err := srv.CreateTaskRun(ctx, &pb.CreateTaskRunRequest{
+		TaskRun: &pb.TaskRun{
+			ApiVersion: "v1beta1",
+			Metadata: &pb.ObjectMeta{
+				Uid:       "31415926",
+				Name:      "mytaskrun",
+				Namespace: "default",
+			}}})
+	if err != nil {
+		t.Fatalf("could not create taskrun: %v", err)
+	}
+	t.Logf("Created taskrun: %s", r.String())
+
+	// Test if we can find inserted taskrun
+	res, err := srv.GetTaskRun(ctx, &pb.GetTaskRunRequest{Uid: r.GetMetadata().GetUid()})
+	if err != nil || !cmp.Equal(r.String(), res.String()) {
```

Comment:
I'd move the cmp and err comparison to separate lines so that it is easier to debug what went wrong if the test fails. Also, you should use `cmp.Diff` to compare complex objects like structs. That way, if the test fails, the output will contain the diff between what we wanted and we we got. See the example in https://godoc.org/github.com/google/go-cmp/cmp#Diff

Reply:
done